android/app/src/main/java/com/mattermost/rnbeta/RNPasteableEditTextOnPasteListener.java

```diff
@@ -0,0 +1,117 @@
+package com.mattermost.rnbeta;
+
+import android.content.ClipData;
+import android.content.ClipboardManager;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.res.AssetFileDescriptor;
+import android.net.Uri;
+import android.util.Patterns;
+import android.webkit.MimeTypeMap;
+import android.webkit.URLUtil;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.WritableArray;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.uimanager.events.RCTEventEmitter;
+import com.mattermost.share.RealPathUtil;
+
+import java.io.FileNotFoundException;
+import java.util.regex.Matcher;
+
+public class RNPasteableEditTextOnPasteListener implements RNEditTextOnPasteListener {
+
+    private RNPasteableEditText mEditText;
+
+    RNPasteableEditTextOnPasteListener(RNPasteableEditText editText) {
+        mEditText = editText;
+    }
+
+    @Override
+    public void onPaste(Uri itemUri) {
+        ReactContext reactContext = (ReactContext)mEditText.getContext();
+        String uri = itemUri.toString();
+
+        WritableArray images = null;
+        WritableMap error = null;
+
+        // Special handle for Google docs
+        if (uri.equals("content://com.google.android.apps.docs.editors.kix.editors.clipboard")) {
+            ClipboardManager clipboardManager = (ClipboardManager) reactContext.getSystemService(Context.CLIPBOARD_SERVICE);
+            ClipData clipData = clipboardManager.getPrimaryClip();
+            if (clipData == null) {
+                return;
+            }
+
+            ClipData.Item item = clipData.getItemAt(0);
+            String htmlText = item.getHtmlText();
+            // Find uri from html
+            Matcher matcher = Patterns.WEB_URL.matcher(htmlText);
+            if (matcher.find()) {
+                uri = htmlText.substring(matcher.start(1), matcher.end());
+            }
+        }
+
+        if (uri.startsWith("http")) {
+            Thread pastImageFromUrlThread = new Thread(new RNPasteableImageFromUrl(reactContext, mEditText, uri));
+            pastImageFromUrlThread.start();
+            return;
+        }
+
+        uri = RealPathUtil.getRealPathFromURI(reactContext, itemUri);
```

Comment:
I've installed Microsoft one note to copy and paste images and this line here is crashing the app

```
2019-09-07 04:01:36.424 8889-8889/com.mattermost.rnbeta E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.mattermost.rnbeta, PID: 8889
    android.os.BadParcelableException: java.lang.NullPointerException: Attempt to invoke virtual method 'byte[] java.lang.String.getBytes()' on a null object reference
        at android.os.Parcel.createException(Parcel.java:1944)
        at android.os.Parcel.readException(Parcel.java:1910)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
        at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:146)
        at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:698)
        at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1458)
        at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1295)
        at android.content.ContentResolver.openFileDescriptor(ContentResolver.java:1148)
        at android.content.ContentResolver.openFileDescriptor(ContentResolver.java:1102)
        at com.mattermost.share.RealPathUtil.getPathFromSavingTempFile(RealPathUtil.java:125)
        at com.mattermost.share.RealPathUtil.getRealPathFromURI(RealPathUtil.java:87)
        at com.mattermost.rnbeta.RNPasteableEditTextOnPasteListener.onPaste(RNPasteableEditTextOnPasteListener.java:62)
        at com.mattermost.rnbeta.RNPasteableActionCallback.onActionItemClicked(RNPasteableActionCallback.java:46)
        at android.widget.Editor$TextActionModeCallback.onActionItemClicked(Editor.java:4175)
        at com.android.internal.policy.DecorView$ActionModeCallback2Wrapper.onActionItemClicked(DecorView.java:2392)
        at com.android.internal.view.FloatingActionMode$3.onMenuItemSelected(FloatingActionMode.java:98)
        at com.android.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:776)
        at com.android.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:148)
        at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:923)
        at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:913)
        at com.android.internal.view.FloatingActionMode.lambda$setFloatingToolbar$0(FloatingActionMode.java:123)
        at com.android.internal.view.-$$Lambda$FloatingActionMode$LU5MpPuKYDtwlFAuYhXYfzgLNLE.onMenuItemClick(Unknown Source:2)
        at com.android.internal.widget.FloatingToolbar$FloatingToolbarPopup$2.onClick(FloatingToolbar.java:443)
        at android.view.View.performClick(View.java:6597)
        at android.view.View.performClickInternal(View.java:6574)
        at android.view.View.access$3100(View.java:778)
        at android.view.View$PerformClick.run(View.java:25885)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
2019-09-07 04:01:36.435 8889-8922/com.mattermost.rnbeta W/ReactNativeJS: Handling native error "android.os.BadParcelableException: java.lang.NullPointerException: Attempt to invoke virtual method 'byte[] java.lang.String.getBytes()' on a null object reference\n\tat android.os.Parcel.createException(Parcel.java:1944)\n\tat android.os.Parcel.readException(Parcel.java:1910)\n\tat android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)\n\tat android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:146)\n\tat android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:698)\n\tat android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:1458)\n\tat android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1295)\n\tat android.content.ContentResolver.openFileDescriptor(ContentResolver.java:1148)\n\tat android.content.ContentResolver.openFileDescriptor(ContentResolver.java:1102)\n\tat com.mattermost.share.RealPathUtil.getPathFromSavingTempFile(RealPathUtil.java:125)\n\tat com.mattermost.share.RealPathUtil.getRealPathFromURI(RealPathUtil.java:87)\n\tat com.mattermost.rnbeta.RNPasteableEditTextOnPasteListener.onPaste(RNPasteableEditTextOnPasteListener.java:62)\n\tat com.mattermost.rnbeta.RNPasteableActionCallback.onActionItemClicked(RNPasteableActionCallback.java:46)\n\tat android.widget.Editor$TextActionModeCallback.onActionItemClicked(Editor.java:4175)\n\tat com.android.internal.policy.DecorView$ActionModeCallback2Wrapper.onActionItemClicked(DecorView.java:2392)\n\tat com.android.internal.view.FloatingActionMode$3.onMenuItemSelected(FloatingActionMode.java:98)\n\tat com.android.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:776)\n\tat com.android.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:148)\n\tat com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:923)\n\tat com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:913)\n\tat com.android.internal.view.FloatingActionMode.lambda$setFloatingToolbar$0(FloatingActionMode.java:123)\n\tat com.android.internal.view.-$$Lambda$FloatingActionMode$LU5MpPuKYDtwlFAuYhXYfzgLNLE.onMenuItemClick(Unknown Source:2)\n\tat com.android.internal.widget.FloatingToolbar$FloatingToolbarPopup$2.onClick(FloatingToolbar.java:443)\n\tat android.view.View.performClick(View.java:6597)\n\tat android.view.View.performClickInternal(View.java:6574)\n\tat android.view.View.access$3100(View.java:778)\n\tat android.view.View$PerformClick.run(View.java:25885)\n\tat android.os.Handler.handleCallback(Handler.java:873)\n\tat android.os.Handler.dispatchMessage(Handler.java:99)\n\tat android.os.Looper.loop(Looper.java:193)\n\tat android.app.ActivityThread.main(ActivityThread.java:6669)\n\tat java.lang.reflect.Method.invoke(Native Method)\n\tat com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)\n\tat com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)\n"
```

Reply:
fixed in this commit https://github.com/mattermost/mattermost-mobile/pull/3084/commits/ffa49ec11fb8492969f5bfb1bc7b125a38734466